tests/verilog/typedef_across_files.ys

```diff
@@ -7,12 +7,15 @@ typedef T [3:0] S;
 EOF
 
 read_verilog -sv <<EOF
+// redefine T, which also affects the previous definition of S
+typedef logic [7:0] T;
+
 module top;
     T t;
     S s;
     always @* begin
-        assert ($bits(t) == 1);
-        assert ($bits(s) == 4);
+        assert ($bits(t) == 8);
+        assert ($bits(s) == 8);
```

Comment:
I'm no verilog person, is this expected behaviour? I'd expect it to be 32.

Reply:
Yosys doesn't handle multiple packed dimensions, so that's why you're seeing `8` in both cases. However, I believe identifiers within a typedef should be resolved at the site of their usage, rather than from the scope in which that typedef is instantiated, so the later re-definition of `T` shouldn't affect `S`. Further, this arguably loses the existing/previous test coverage, which isn't ideal.

The way globals are currently handled is fundamentally flawed. Modules should indeed be able to shadow typedefs, parameters, etc., without affecting the definition of other globals which depend on the shadowed parameters. In this sense, global vs. module scope is no different then outer vs. inner generate block scoping.

Reply:
I dropped the globals redefinition issue for now and only kept the original commit (struct member types not searching up the type scope stack), I'll open an issue about the former.